src/hotspot/os/linux/os_linux.cpp

```diff
@@ -2266,6 +2266,19 @@ bool os::Linux::print_container_info(outputStream* st) {
   print_container_helper(st, OSContainer::memory_usage_in_bytes(), "memory_usage_in_bytes");
   print_container_helper(st, OSContainer::memory_max_usage_in_bytes(), "memory_max_usage_in_bytes");
 
+  // the kmem (kernel memory) values are only available in cgroupv1
+  if (p_ct != NULL && strcmp(p_ct, "cgroupv1") == 0) {
+    print_container_helper(st, OSContainer::kernel_memory_usage_in_bytes(), "kernel_memory_usage_in_bytes");
```

Comment:
We wouldn't need the extra version checking gymnastics if we'd call an `OSContainer::print_version_specific_info(st);` here instead.

Reply:
sounds like a good idea, but with the kmem stuff deprecated (Kernel commit adding the deprecation:
https://github.com/torvalds/linux/commit/0158115f702b )  we should maybe wait first for a reply from Thomas why he is still interested in it.

Reply:
OK.

Reply:
Hi Severin, I think the OSContainer::print_version_specific_info(st) sounds like a good idea. It would mean that we keep the API smaller and do not overload it with stuff that is in fact only available in v1  (or only available in v2).